### PR TITLE
tag images with sha

### PR DESCRIPTION
**Pull request by markgllin**
tag images with sha instead of allowing it to be manually specified

### CODE REVIEW

**Review by davegallant — APPROVED** [review]

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -52,6 +50,14 @@ jobs:
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_PAT }}
 
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: "ghcr.io/${{ inputs.docker_image_repository }}"
+          tags: |
+            type=sha,enable=true,priority=100,prefix=sha-,suffix=,format=short
```
Comment (davegallant) [inline .github/workflows/build-and-push.yml:59]: This should also tag latest

**Review by davegallant — CHANGES_REQUESTED** [review]
Need to tag the latest.

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -52,6 +50,14 @@ jobs:
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_PAT }}
 
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: "ghcr.io/${{ inputs.docker_image_repository }}"
+          tags: |
+            type=sha,value=latest,enable=true,priority=100,prefix=sha-,suffix=,format=short
```
Comment (davegallant) [inline .github/workflows/build-and-push.yml:59]:
```suggestion
            type=sha,enable=true,priority=100,prefix=sha-,suffix=,format=short
            type=raw,value=latest
```

Shouldn't it be like this? 

https://github.com/docker/metadata-action/blob/b2391d37b4157fa4aa2e118d643f417910ff3242/README.md

Comment (markgllin) [reply .github/workflows/build-and-push.yml:59]: ooohh i misunderstood the doc to mean that only for branches based on their example

**Review by davegallant — APPROVED** [review]

**Pull request merged**